frontend/packages/dev-console/src/components/monitoring/metrics/MetricsQueryInput.tsx

```diff
@@ -2,66 +2,121 @@ import * as React from 'react';
 import { connect } from 'react-redux';
 import * as fuzzy from 'fuzzysearch';
 import { RootState } from '@console/internal/redux';
+import { Button } from '@patternfly/react-core';
 import { Dropdown } from '@console/internal/components/utils';
 import { queryBrowserRunQueries, queryBrowserPatchQuery } from '@console/internal/actions/ui';
 import { getActiveNamespace } from '@console/internal/reducers/ui';
+import { QueryInput } from '@console/internal/components/monitoring/metrics';
 import { QueryObj } from '@console/internal/components/monitoring/query-browser';
 import { metricsQuery, getTopMetricsQueries } from '../queries';
 import './MetricsQueryInput.scss';
 
 const ADD_NEW_QUERY = '#ADD_NEW_QUERY#';
+const CUSTOM_QUERY = 'Custom Query';
 
 type StateProps = {
   namespace: string;
+  text: string;
+  query: string;
 };
 
 type MetricsQueryInputProps = {
   patchQuery: (patch: QueryObj) => void;
-  runQueries: () => never;
+  runQueries: () => void;
   namespace: string;
+  text?: string;
+  query?: string;
 };
 
 export const MetricsQueryInput: React.FC<MetricsQueryInputProps> = ({
   patchQuery,
   runQueries,
   namespace,
+  text,
+  query,
 }) => {
   const items = metricsQuery;
-  const autocompleteFilter = (text, item) => fuzzy(text, item);
+  const autocompleteFilter = (strText, item) => fuzzy(strText, item);
   const defaultActionItem = [
     {
-      actionTitle: `Custom Query`,
+      actionTitle: CUSTOM_QUERY,
       actionKey: ADD_NEW_QUERY,
     },
   ];
 
+  const [title, setTitle] = React.useState('Select Query');
   const [metric, setMetric] = React.useState('');
+  const [showPromQl, setShowPromQl] = React.useState(false);
+  const [isPromQlDisabled, setIsPromQlDisabled] = React.useState(false);
   React.useEffect(() => {
     if (metric && metric !== ADD_NEW_QUERY) {
-      const query = getTopMetricsQueries(namespace)[metric];
-      patchQuery({ text: query });
+      const queryMetrics = getTopMetricsQueries(namespace)[metric];
+      patchQuery({ text: queryMetrics });
       runQueries();
     }
   }, [metric, namespace, patchQuery, runQueries]);
-  const onChange = (selVal: string) => {
-    setMetric(selVal);
+
+  React.useEffect(() => {
+    if (text.localeCompare(query) !== 0) {
```

Comment:
This will throw an error if the text is undefined.

![Screenshot from 2020-01-22 08-04-25](https://user-images.githubusercontent.com/2561818/72860796-f5faee00-3ced-11ea-9d45-df663e6c6254.png)

Reply:
@vikram-raj thanks, have added check PTAL